pkg/credentials/credentials.go

```diff
@@ -59,99 +75,139 @@ type ProviderEnvironmentVariable struct {
-			{Name: "HCLOUD_TOKEN", MachineControllerName: "HZ_TOKEN"},
+		return f.parseCredentialVariables([]ProviderEnvironmentVariable{
+			{Name: HetznerTokenKey, MachineControllerName: HetznerTokenKeyMC},
 		}, defaultValidationFunc)
 	case kubeone.CloudProviderNameDigitalOcean:
-		return parseCredentialVariables([]ProviderEnvironmentVariable{
-			{Name: "DIGITALOCEAN_TOKEN", MachineControllerName: "DO_TOKEN"},
+		return f.parseCredentialVariables([]ProviderEnvironmentVariable{
+			{Name: DigitalOceanTokenKey, MachineControllerName: DigitalOceanTokenKeyMC},
 		}, defaultValidationFunc)
 	case kubeone.CloudProviderNameGCE:
-		gsa, err := parseCredentialVariables([]ProviderEnvironmentVariable{
-			{Name: "GOOGLE_CREDENTIALS", MachineControllerName: "GOOGLE_SERVICE_ACCOUNT"},
+		gsa, err := f.parseCredentialVariables([]ProviderEnvironmentVariable{
+			{Name: GoogleServiceAccountKey, MachineControllerName: GoogleServiceAccountKeyMC},
 		}, defaultValidationFunc)
 		if err != nil {
 			return nil, errors.WithStack(err)
 		}
 		// encode it before sending to secret to be consumed by
 		// machine-controller, as machine-controller assumes it will be double encoded
-		gsa["GOOGLE_SERVICE_ACCOUNT"] = base64.StdEncoding.EncodeToString([]byte(gsa["GOOGLE_SERVICE_ACCOUNT"]))
+		gsa[GoogleServiceAccountKeyMC] = base64.StdEncoding.EncodeToString([]byte(gsa[GoogleServiceAccountKeyMC]))
 		return gsa, nil
 	case kubeone.CloudProviderNamePacket:
-		return parseCredentialVariables([]ProviderEnvironmentVariable{
-			{Name: "PACKET_AUTH_TOKEN", MachineControllerName: PacketAPIKey},
+		return f.parseCredentialVariables([]ProviderEnvironmentVariable{
+			{Name: PacketAPIKey, MachineControllerName: PacketAPIKeyMC},
 			{Name: PacketProjectID},
 		}, defaultValidationFunc)
 	case kubeone.CloudProviderNameVSphere:
-		vscreds, err := parseCredentialVariables([]ProviderEnvironmentVariable{
-			{Name: "VSPHERE_SERVER", MachineControllerName: VSphereAddress},
-			{Name: "VSPHERE_USER", MachineControllerName: VSphereUsername},
+		vscreds, err := f.parseCredentialVariables([]ProviderEnvironmentVariable{
+			{Name: VSphereAddress, MachineControllerName: VSphereAddressMC},
+			{Name: VSphereUsername, MachineControllerName: VSphereUsernameMC},
 			{Name: VSpherePassword},
 		}, defaultValidationFunc)
 		if err != nil {
 			return nil, errors.WithStack(err)
 		}
 
 		// force scheme, as machine-controller requires it while terraform does not
-		vscreds[VSphereAddress] = "https://" + vscreds[VSphereAddress]
+		vscreds[VSphereAddressMC] = "https://" + vscreds[VSphereAddressMC]
 		return vscreds, nil
 	}
 
 	return nil, errors.New("no provider matched")
 }
 
-func parseCredentialVariables(envVars []ProviderEnvironmentVariable, validationFunc func(map[string]string) error) (map[string]string, error) {
+func parseAWSCredentials() (map[string]string, error) {
+	creds := make(map[string]string)
+	envCredsProvider := credentials.NewEnvCredentials()
+	envCreds, err := envCredsProvider.Get()
```

Comment:
Judging by [aws source](https://github.com/aws/aws-sdk-go/blob/034e9fbdb3877e1cf27f02f2ea38f50927b405c6/aws/credentials/env_provider.go#L55) this will error out in case when ENVs are not found, and workflow will not fallback down to shared credentials.